Issue with autoPagingIterable() and RequestOptions

(Using stripe-java 1.38 and gson 2.2.4.)

This code:
```java
RequestOptions options = RequestOptions.builder().setApiKey("sk_test_...").build();

Map<String, Object> planParams = new HashMap<String, Object>();
planParams.put("limit", 2);

Iterable<Plan> itPlans = Plan.list(planParams, options).autoPagingIterable();

for (Plan plan : itPlans) {
    System.out.println(plan.getId());
}
```

will list the first 4 plans, then fail with the following exception:
```
java.lang.RuntimeException: Unable to lazy-load stripe objects
	at com.stripe.model.PagingIterator.next(PagingIterator.java:60)
	at com.stripe.model.PagingIterator.next(PagingIterator.java:12)
	at StripeExample.main(StripeExample.java:40)
Caused by: com.stripe.exception.AuthenticationException: No API key provided. (HINT: set your API key using 'Stripe.apiKey = <API-KEY>'. You can generate API keys from the Stripe web interface. See https://stripe.com/api for details or email support@stripe.com if you have questions.
	at com.stripe.net.LiveStripeResponseGetter._request(LiveStripeResponseGetter.java:409)
	at com.stripe.net.LiveStripeResponseGetter.request(LiveStripeResponseGetter.java:65)
	at com.stripe.net.APIResource.request(APIResource.java:131)
	at com.stripe.model.PagingIterator.list(PagingIterator.java:83)
	at com.stripe.model.PagingIterator.next(PagingIterator.java:55)
	... 2 more
```

Strangely enough, it works flawlessly when using:
```java
planParams.put("limit", 1);
```
instead.

There is also no issue if the API key is set at the global level:
```java
Stripe.apiKey = "sk_test_..."
```

--- a/src/main/java/com/stripe/model/PagingIterator.java
+++ b/src/main/java/com/stripe/model/PagingIterator.java
@@ -80,6 +80,6 @@ public class PagingIterator<T extends HasId> extends APIResource implements Iter
 		final Map<String, Object> params, 
 		final RequestOptions options
 	) throws Exception {
-		return APIResource.request(RequestMethod.GET, url, params, collectionType, options);
+		return APIResource.requestCollection(url, params, collectionType, options);
 	}
 }

